### PR TITLE
JPERF-470: Let JVM exit

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

**Comment by dagguh** [comment]
I found a non-daemon thread created by `sun.net.httpserver.ServerImpl#start`:
```java
new Thread(this.dispatcher)
```
I'll start the entire server in a daemon thread, because the injected `Executor` is ignored in this case.

**Comment by dagguh** [comment]
Weird, it passes locally for me

**Comment by dagguh** [comment]
CircleCI runs old JDK
https://circleci.com/gh/dagguh/virtual-users/88 :
```
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
```

It's hard to bump it (because we use the `machine` executor, not Dockerised). Tried bumping the image here: https://circleci.com/gh/dagguh/virtual-users/89, still old JDK.

Perhaps this shows the real problem - the tests are very highly dependent on JDK internals - `sun.net.httpserver` and will pass/fail based on the contributor environment. I'll replace this with a more stable and predictable embedded HTTP server.

**Comment by dagguh** [comment]
I tried https://github.com/NanoHttpd/nanohttpd : https://github.com/dagguh/virtual-users/commit/2c752866f9a89fc28cd177e66f6723e3942f1066
It was pretty bad: it started reading HTTP request body as a HTTP status line (perhaps across separate requests).
I switched to Tomcat and it works much better, although with a bigger shutdown overhead (~3 sec).

**Comment by dagguh** [comment]
It was [green on a side branch](https://circleci.com/gh/dagguh/virtual-users/93), now it's red after a squash.
https://circleci.com/gh/atlassian/virtual-users :
Sometimes green takes 14 minutes, sometimes 29 minutes 🤔

**Comment by dagguh** [comment]
A rerun on the branch that was green before also fails: https://circleci.com/gh/dagguh/virtual-users/96
It's flaky. I'll SSH-debug it.

**Comment by dagguh** [comment]
I've got the local repro in Docker reduced to 2 CPUs.

**Comment by dagguh** [comment]
Local is flaky: 2/3 is red. I debugged it and saw that the reported threads disappear *almost* instantly after they're done running `LoadSegment.close`. Almost. I'll try to wait for them to disappear with a short sleep.

**Comment by dagguh** [comment]
Now it hangs the gradle task, there's no stdout and circleci times out.
This absolutely does not happen locally. I have no clue why CI is different than local 😩

**Comment by dagguh** [comment]
`failFast = true` helped. At least now I can see the artifacts: https://circleci.com/gh/dagguh/virtual-users/105

```
com.atlassian.performance.tools.virtualusers.LoadTestTerminationIT > shouldCloseAFastBrowser FAILED
    java.lang.AssertionError: 
    Expecting:
     <PT5M47.819S>
    to be less than:
     <PT5S> 
```

Over 5 minutes of unpexpected overhead? Snap!
Artifacts show a PT5M45S gap at the very start:
```
2019-04-26T16:22:04,423Z DEBUG parent-for-shouldCloseAFastBrowser [] [com.atlassian.performance.tools.virtualusers.LoadTest] Skipped the setup
2019-04-26T16:27:49,513Z INFO  virtual-user-0 [applying load #6ba229d3-593d-4ec2-8904-fbf910a75582] [com.atlassian.performance.tools.virtualusers.LoadTest] Waiting for PT0S
```

**Comment by dagguh** [comment]
It's proper red-green now. The GH GUI still sorts it by date. I can rebase it once again with fixed dates. LMK if I should.

File: build.gradle.kts [hunk 1]
```diff
@@ -129,6 +129,7 @@ tasks.getByName("test", Test::class).apply {
 
 val testIntegration = task<Test>("testIntegration") {
     include("**/*IT.class")
+    failFast = true
```
Comment (wyrzyk) [inline build.gradle.kts]: 👍

**Pull request merged**